### ISSUE TITLE
Checkpoint sync for Eth testnets

### ISSUE THREAD

**Issue by jgresham** [NiceNode/nice-node#451]
## Feature details

- [ ] Use checkpoint sync for Sepolia and Holesky for Consensus Clients that already use it for mainnet
- [ ] Test it because some Consensus clients don't work with certain URLs 

[Checkpoint sync urls by network](https://eth-clients.github.io/checkpoint-sync-endpoints/#holesky)

**Comment by arionrefat** [comment]
@jgresham Hi, I want to work on it. Can you more pointers on how can I solve this ?

**Comment by jgresham** [comment]
> [@jgresham](https://github.com/jgresham) Hi, I want to work on it. Can you more pointers on how can I solve this ?

great! The work is mostly done in another repo, which serves each node config controls to NiceNode, and might just be setting the urls in the json files. You can run this api locally and have NiceNode use the local server to test. The format of these json files isn't really documented well however. A couple options: 
1. set the `checkpointSyncUrl: <url>` in the Ethereum json file under 
```
Sepolia config...
    "serviceId": "consensusClient", 
    "configValues": {
        "checkpointSyncUrl: <sepolia-checkpoint-sync>
```
https://github.com/NiceNode/controller-api/blob/main/src/app/api/controller/NodeSpecs/ethereum/ethereum-v1.0.0.json#L118 

This should set the `checkpointSyncUrl` for each consensus client.
Also adding `"addNodeFlow": "optional"` to the `checkpointSyncUrl: { }` config for each consensus client should allow the user to see this when adding the node and before starting the node for the first time (important for checkpoint sync).

2. we could just add on the checkpoint sync flag for each network config like here https://github.com/NiceNode/controller-api/blob/main/src/app/api/controller/NodeSpecs/lodestar/lodestar-v1.0.0.json#L67

**Comment by jgresham** [comment]
Sometimes the checkpoint sync servers are slow or the local client doesn't wait long enough. Adjusting the timeout config for the client might be necessary (lighthouse docs https://lighthouse-book.sigmaprime.io/checkpoint-sync.html#adjusting-the-timeout)

**Comment by arionrefat** [comment]
@jgresham  Thank you so much man. I will look into it and ask questions if require.

**Comment by arionrefat** [comment]
> we could just add on the checkpoint sync flag for each network config like here https://github.com/NiceNode/controller-api/blob/main/src/app/api/controller/NodeSpecs/lodestar/lodestar-v1.0.0.json#L67

![Image](https://github.com/user-attachments/assets/a777837b-c659-4cce-bbd6-a9af3da88e9f)

I went with option 2 and tested it out in nice-node

![Image](https://github.com/user-attachments/assets/566ddfb3-2dcb-431a-bc64-065d8e9b71e2)

it seems it's synchronising for Sepolia and Holesky for Consensus Clients.

Am I doing it right? @jgresham

**Comment by jgresham** [comment]
> > we could just add on the checkpoint sync flag for each network config like here https://github.com/NiceNode/controller-api/blob/main/src/app/api/controller/NodeSpecs/lodestar/lodestar-v1.0.0.json#L67
> 
> ![Image](https://github.com/user-attachments/assets/a777837b-c659-4cce-bbd6-a9af3da88e9f)
> 
> I went with option 2 and tested it out in nice-node
> 
> ![Image](https://github.com/user-attachments/assets/566ddfb3-2dcb-431a-bc64-065d8e9b71e2)
> 
> it seems it's synchronising for Sepolia and Holesky for Consensus Clients.
> 
> Am I doing it right? [@jgresham](https://github.com/jgresham)

It should be working. You could verify by comparing the latest block or slot shown in NiceNode or the logs of each client to a block explorer like https://sepolia.etherscan.io/ or https://sepolia.beaconcha.in/

**Comment by jgresham** [comment]
I suppose you should also see some confirmation of checkpoint sync working in the client logs when you start the node for the first time. After the node is synced it no longer uses checkpoint sync

**Comment by arionrefat** [comment]
Alright, I will check it out. Thank you

**Comment by arionrefat** [comment]
Yeah it seems to be it's working, the nodes are running. Where do I sent the PR then? https://github.com/NiceNode/controller-api here?

**Comment by jgresham** [comment]
> Yeah it seems to be it's working, the nodes are running. Where do I sent the PR then? https://github.com/NiceNode/controller-api here?

yeah send the PR there please!

**Comment by arionrefat** [comment]
@jgresham Here is the PR : https://github.com/NiceNode/controller-api/pull/18

**Comment by jgresham** [comment]
> [@jgresham](https://github.com/jgresham) Here is the PR : [NiceNode/controller-api#18](https://github.com/NiceNode/controller-api/pull/18)

Added checkpoint urls for other consensus clients that we have checkpoint sync support for and bumped the controller versions https://github.com/NiceNode/controller-api/pull/19

You can see here that it is deployed now! https://api.nicenode.xyz/api/controller/lodestar-beacon

**Comment by arionrefat** [comment]
@jgresham thank you man. I really appreciate your help, it's really nice working with you. It would be nice if we can get connected, followed you on X.
